### PR TITLE
Record soft failure bsc#1004643 & bsc#1111483

### PR DESCRIPTION
**Pull request by jknphy**
Record soft failure for bsc#1004643.
- Related ticket: https://progress.opensuse.org/issues/42782
- Needles: [~needles sle~](https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/990)
- Verification run: Today due to low-performance in the ppc worker I cannot get verification (it is constantly failing in different steps of the test suite), but from last day I have [sle-15-SP1-extra_tests_on_gnome@ppc64le](http://dhcp42.suse.cz/tests/855#step/yast2_lan_restart_devices/31) to see the second soft-failure (as the first will be only visible in prod sporadically).

### CODE REVIEW

File: tests/x11/yast2_lan_restart_devices.pm [hunk 1]
```diff
@@ -17,6 +17,15 @@ use strict;
 use testapi;
 use y2lan_restart_common qw(initialize_y2lan open_network_settings close_network_settings check_network_status);
 
+sub record_soft_failure_bsc1111483 {
+    if (match_has_tag('yast2_lan_device_workaround-bsc1004643')) {
```
Comment (okurz) [inline tests/x11/yast2_lan_restart_devices.pm]: Would you mind an early return?

```suggestion
    return 0 unless match_has_tag('yast2_lan_device_workaround-bsc1004643');
```

File: tests/x11/yast2_lan_restart_devices.pm [hunk 2]
```diff
@@ -79,37 +88,44 @@ sub add_device {
 
 sub select_special_device_tab {
     my $device = shift;
+
```
Comment (okurz) [inline tests/x11/yast2_lan_restart_devices.pm]: these whitespace changes seem unrelated. I suggest to revert. If you want to change the style I suggest to change the perltidy file or suggest a change valid for all test modules.

File: tests/x11/yast2_lan_restart_devices.pm [hunk 3]
```diff
@@ -79,37 +88,44 @@ sub add_device {
...
     open_network_settings;
     send_key 'tab';
     send_key 'tab';
     send_key 'home';
-    send_key_until_needlematch "yast2_lan_device_${device}_selected", 'down';
-    send_key 'alt-i';                     # Edit NIC
+    send_key_until_needlematch ["yast2_lan_device_${device}_selected", "yast2_lan_device_workaround-bsc1004643"], 'down', 5;
+    if (record_soft_failure_bsc1111483) {
```
Comment (okurz) [inline tests/x11/yast2_lan_restart_devices.pm]: similar to above, how about `return if record_soft_failure_bsc1111483;` just to save some space? :)

Comment (jknphy) [reply tests/x11/yast2_lan_restart_devices.pm]: I had it like that at the begining, but Travis didn't pass, it is detected as duplicated code. Neither worked `return if (record_soft_failure_bsc1111483)` nor `return if (record_soft_failure_bsc1111483()).

File: tests/x11/yast2_lan_restart_devices.pm (hunk 2 above)
Comment (jknphy) [reply tests/x11/yast2_lan_restart_devices.pm]: I modified the function/file so I guess I can apply my own style as there is not rigid convention about this thing AFIK and in the repo there are many example of both styles. For me the point to use the space is that in Perl there is not signature of the functions so it is much more clear separate the parameters from the rest of the code.

File: tests/x11/yast2_lan_restart_devices.pm (hunk 3 above)
Comment (jknphy) [reply tests/x11/yast2_lan_restart_devices.pm]: @okurz thanks for the tip in the call, no duplicate, it is the name of the function likely, which our make rule was identifying it wrongly. Let's see if it works changing it.

File: lib/y2lan_restart_common.pm [hunk 4]
```diff
@@ -47,7 +47,11 @@ sub initialize_y2lan {
 sub open_network_settings {
     type_string "yast2 lan\n";
     accept_warning_network_manager_default;
-    assert_screen 'yast2_lan', 100;       # yast2 lan overview tab
+    assert_screen([qw(yast2_lan yast2_lan-bsc1004643)], 100);    # yast2 lan overview tab
```
Comment (rwx788) [inline lib/y2lan_restart_common.pm]: As we have discussed in person, let's simply create workaround needle with bug reference and press home unconditionally, so test module works with multiple interfaces. Other thing we need `send_key_until_needle_match` to select correct one. WDYT?

Comment (jknphy) [reply lib/y2lan_restart_common.pm]: Yes, right! we avoid to remove the soft-failure as in this case to send key home is not specific from the workaround and the needle is enough. I changed the needles to yast2_lan to match as we created initially.

File: lib/y2lan_restart_common.pm [hunk 5]
```diff
@@ -48,6 +48,7 @@ sub open_network_settings {
     type_string "yast2 lan\n";
     accept_warning_network_manager_default;
     assert_screen 'yast2_lan', 100;       # yast2 lan overview tab
+    send_key 'home';                      # select first device
```
Comment (rwx788) [inline lib/y2lan_restart_common.pm]: Still, I'm not sure if it guarantees selecting correct device, except the identified case. But I would really make it work, so we don't have to touch it if ever enable this test on bare-metal machine.

File: tests/x11/yast2_lan_restart_devices.pm [hunk 6]
```diff
@@ -17,11 +17,18 @@ use strict;
 use testapi;
 use y2lan_restart_common qw(initialize_y2lan open_network_settings close_network_settings check_network_status);
 
+sub check_bsc1111483 {
```
Comment (okurz) [inline tests/x11/yast2_lan_restart_devices.pm]: :+1: :)

**Comment by okurz** [comment]
LGTM. I will leave the rest to @rwx788 as I am not sure if he wants that change to be done as part of this PR

File: lib/y2lan_restart_common.pm (hunk 5 above)
Comment (jknphy) [reply lib/y2lan_restart_common.pm]: Without seeing other scenarios it is hard to say, this is based on the assumption that the first would be the right one, any other strategy like assert_and_click  would require create needle for each product to click on the corresponding interface (if scroll is not needed). We can always deal with more specific bare-metal scenarios in future, I wanted to cover only ppc here (which already has a lot of problem and it is time-consuming to get correct verification).

Comment (rwx788) [reply lib/y2lan_restart_common.pm]: Alright, let's leave it for us in the future =)

**Comment by rwx788** [comment]
@jknphy: Please, take a look on osd and o3 if this works.
